src/modules/service-provider/domain/service-provider.service.ts

```diff
@@ -23,4 +42,117 @@ export class ServiceProviderService {
 
         return Array.from(serviceProviders.values());
     }
+
+    public async updateServiceProvidersForVidis(): Promise<void> {
+        this.logger.info('Aktualisierung der ServiceProvider für VIDIS-Angebote wurde gestartet.');
+
+        const vidisKeycloakGroup: string = this.vidisConfig.KEYCLOAK_GROUP;
+        const vidisKeycloakRole: string = this.vidisConfig.KEYCLOAK_ROLE;
+        const vidisRegionName: string = this.vidisConfig.REGION_NAME;
+        const schulstrukturknoten: string = this.organisationRepo.ROOT_ORGANISATION_ID;
+
+        const vidisAngebote: VidisAngebot[] = await this.vidisService.getActivatedAngeboteByRegion(vidisRegionName);
+
+        const allMappingsBeenDeleted: boolean = await this.organisationServiceProviderRepo.deleteAll();
+        if (allMappingsBeenDeleted)
+            this.logger.info('All mappings between Organisation and ServiceProvider were deleted.');
+
+        await Promise.allSettled(
+            vidisAngebote.map(async (angebot: VidisAngebot) => {
+                const existingServiceProvider: Option<ServiceProvider<true>> =
+                    await this.serviceProviderRepo.findByName(angebot.angebotTitle);
+
+                const angebotLogoMediaType: string = this.determineMediaTypeFor(angebot.angebotLogo);
+
+                let serviceProvider: ServiceProvider<false>;
+                if (existingServiceProvider) {
+                    serviceProvider = ServiceProvider.construct(
+                        existingServiceProvider.id,
+                        existingServiceProvider.createdAt,
+                        existingServiceProvider.updatedAt,
+                        angebot.angebotTitle,
+                        ServiceProviderTarget.URL,
+                        angebot.angebotLink,
+                        ServiceProviderKategorie.ANGEBOTE,
+                        schulstrukturknoten,
+                        Buffer.from(angebot.angebotLogo, 'base64'),
+                        angebotLogoMediaType,
+                        vidisKeycloakGroup,
+                        vidisKeycloakRole,
+                        ServiceProviderSystem.NONE,
+                        false,
+                    );
+                    this.logger.info(`ServiceProvider for VIDIS Angebot '${serviceProvider.name}' already exists.`);
```

Comment:
Why do we need to construct the service provider when it already exists in the DB?